engine/src/main/java/org/terasology/logic/characters/CharacterMovementSystemUtility.java

```diff
@@ -63,27 +62,27 @@ public void setToState(EntityRef entity, CharacterStateEvent state) {
         movementComp.footstepDelta = state.getFootstepDelta();
         entity.saveComponent(movementComp);
 
-        setPhysicsLocation(entity, state.getPosition());
+        setPhysicsLocation(entity, JomlUtil.from(state.getPosition()));
 
         // set the pitch to the character's gaze entity
-        Quat4f rotation = new Quat4f(0f, TeraMath.DEG_TO_RAD * state.getPitch(), 0f);
+        Quaternionf rotation = new Quaternionf().rotationYXZ(0f, TeraMath.DEG_TO_RAD * state.getPitch(), 0f);
```

Comment:
Could be simplified to:
```suggestion
        Quaternionf rotation = new Quaternionf().rotationX(TeraMath.DEG_TO_RAD * state.getPitch());
```